lib/pbench/server/__init__.py

```diff
@@ -288,11 +245,16 @@ def _get_valid_dir_option(self, env_name: str, section: str, option: str) -> Pat
         Returns:
             A Path directory object.
         """
-        dir_val = self._get_conf(section, option)
+        try:
+            dir_val = self.conf.get(section, option)
+        except (NoOptionError, NoSectionError) as exc:
+            raise BadConfig(str(exc))
+        else:
+            if not dir_val:
+                raise BadConfig(f"option {option} in section {section} is empty")
```

Comment:
The `else` and indented `if` look funny to me.  I would omit the `else`.

Also, I would offer the following alternative instead:
```
        try:
            dir_val = self.conf.get(section, option)
            if not dir_val:
                raise ValueError(f"option {option} in section {section} is empty")
        except (NoOptionError, NoSectionError, ValueError) as exc:
            raise BadConfig(str(exc))
```
This reports all errors via `BadConfig` from a single, common `raise` statement while still capturing the nature of the various triggering failures.

Reply:
I tried to make this a lift-n-shift operation.  I have a follow-on PR (in draft mode) #3192 in which I am working through additional changes.